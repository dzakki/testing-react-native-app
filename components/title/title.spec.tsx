import { render, screen } from '@testing-library/react-native';
import Title from './index';

describe('Title', () => {
  const content = 'Todo App!';

  test('should render content correctly', () => {
    render(<Title>{content}</Title>);
    expect(screen.getByText(content)).toBeVisible();
  });

  test('should the text content in center', () => {
    render(<Title>{content}</Title>);

    expect(screen.getByText(content)).toHaveStyle({
      textAlign: 'center',
    });
  });

  test('should match snapshot', () => {
    render(<Title>{content}</Title>);
    // https://jestjs.io/docs/snapshot-testing
    expect(screen.toJSON()).toMatchSnapshot();
  })
});

