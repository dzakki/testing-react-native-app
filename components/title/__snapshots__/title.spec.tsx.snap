// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should match snapshot 1`] = `
<Text
  style={
    {
      "marginBottom": 16,
      "textAlign": "center",
    }
  }
>
  Todo App!
</Text>
`;
